server/src/main/java/com/ncc/neon/services/ClusterService.java

```diff
@@ -0,0 +1,35 @@
+package com.ncc.neon.services;
+
+import com.ncc.neon.models.queries.ClusterClause;
+import com.ncc.neon.models.results.TabularQueryResult;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+@Component
+public class ClusterService {
+
+    private ClusterClause clusterClause;
+
+    /**
+     * Clusters the results from the query.
+     *
+     * TODO: Currently returns the same results i.e. method is a passthrough right now
+     *
+     * @param tabularQueryResult the result of the unclustered query
+     * @return a mono containing the new clustered results
+     */
+    public Mono<? extends TabularQueryResult> cluster(TabularQueryResult tabularQueryResult) {
+        this.clusterClause = null; // reset cluster clause
+        return Mono.just(tabularQueryResult);
```

Comment:
Making this method return a TabularQueryResult, and then wrapping it in a Mono in the QueryService, would probably make the cluster method easier to unit test.

Reply:
I like that! Done.